src/mesh/mesh_refinement.hpp

```diff
@@ -77,6 +77,8 @@ class MeshRefinement {
   int AddToRefinement(ParArrayND<Real> pvar_cc, ParArrayND<Real> pcoarse_cc);
   int AddToRefinement(FaceField *pvar_fc, FaceField *pcoarse_fc);
 
+  Coordinates_t GetCoarseCoords() const { return coarse_coords; }
```

Comment:
Maybe return a const ref?
```suggestion
  const Coordinates_t &GetCoarseCoords() const { return coarse_coords; }
```

Reply:
I don't think that works--I need `coarse_coords` to go to the GPU so I don't want it to be a reference.